### ISSUE TITLE
[Idea] Automatic backports for elastic/beats

### ISSUE THREAD

**Issue by mtojek** [elastic/beats#15053]
Hi,

I'm afraid that we can't get rid of backporting changes for a longer time, so maybe we'll consider designing a mechanism handling them automatically.

Maybe we can involve some Github actions to trigger `create_pr` scripts after a successful merge operation.

**Comment by exekias** [comment]
This would be something to explore. Are you thinking on a bot?

For completeness other teams/repos (like Kibana) are using https://github.com/sqren/backport, as a friendly interface to do backports

**Comment by mtojek** [comment]
I'm thinking about some automation about Github actions to eliminate any interaction with human at all. If we adopt any functional approach (Lambda like), there is no need to maintain any infrastructure for that.

**Comment by ycombinator** [comment]
++ to this idea! 

@urso and I had similar discussions in the past (and I'm sure others have too before that) wherein a PR could be labeled with the `backport_pending` label and version labels (e.g. `v.8.0.0`, `v.7.6.0`, etc.). Once the PR is merged, the bot would compare the target branch of the PR (typically `master`) with the version labels and setup backport PRs to the corresponding branches automatically. 

Of course, we'd have to address the situation where a backport cannot be made cleanly due to conflicts — perhaps the bot could simply comment on the original PR about such situations so the PR author is notified to handle these situations manually? 

Also, where would the bot create the backport PR branches? Ideally that could be done on the source repo (typically a developer's fork) instead of polluting the main repo with a large number of backport PR branches. But even if we had to create the backport PR branches in the main repo the same bot could be responsible for cleaning these up automatically once the backport PR is merged.

**Comment by urso** [comment]
One pain point with automatic backports is the changelog. It is already broken from time to time due to backports being merged without people checking the backport is not bogus.

Depending on how the bot handles CI, it might not even need to create a PR (assuming CI is run before merging). It could be just a commit to the particular release branch. In general some kind of merge queue would be nice to have. PRs to be merged or backported are handled in a batch. Only if all related CI was succesful will we merge.

**Comment by ycombinator** [comment]
Mentioning for completeness: the Kibana team is using https://github.com/spalger/kibana-pr-bot for their PR automation. Chatting with @spalger, he said:

> you could definitely rip out the "reactors" and replace them with something beats specific

Where, "reactors" refers to https://github.com/spalger/kibana-pr-bot/tree/master/src/reactors.

**Comment by jlind23** [comment]
Backlog grooming: Closing it for now until further activity.
